niap-cc/NIAPSecExample/app/src/main/java/com/android/certifications/niap/CustomSocketFactory.java

```diff
@@ -0,0 +1,562 @@
+    private final SSLSocketFactory delegate;
+    private final boolean compatible;
+
+    public CustomSocketFactory() {
+        this.delegate =SSLCertificateSocketFactory.getDefault(HANDSHAKE_TIMEOUT, null);
+        this.compatible = false;
+    }
+
+    public CustomSocketFactory(SSLSocketFactory delegate) {
+        this.delegate = delegate;
+        this.compatible = false;
+    }
+
+    /**
+     * Make {@link SSLSocket}s that are compatible with outdated servers.
+     *
+     * @param delegate
+     * @param compatible
+     */
+    public CustomSocketFactory(SSLSocketFactory delegate, boolean compatible) {
+        this.delegate = delegate;
+        this.compatible = compatible;
+    }
+
+    @Override
+    public String[] getDefaultCipherSuites() {
+        return delegate.getDefaultCipherSuites();
+    }
+
+    @Override
+    public String[] getSupportedCipherSuites() {
+        return delegate.getSupportedCipherSuites();
+    }
+
+    private Socket makeSocketSafe(Socket socket, String host) {
+        if (socket instanceof SSLSocket) {
+            CustomSSLSocket tempSocket=
+                    new CustomSSLSocket((SSLSocket) socket, compatible);
+
+            if (delegate instanceof SSLCertificateSocketFactory &&
+                    Build.VERSION.SDK_INT>=Build.VERSION_CODES.JELLY_BEAN_MR1) {
+                ((android.net.SSLCertificateSocketFactory)delegate)
+                        .setHostname(socket, host);
+            }
+            else {
+                tempSocket.setHostname(host);
+            }
+
+            socket = tempSocket;
+        }
+        return socket;
+    }
+
+    @Override
+    public Socket createSocket(Socket s, String host, int port, boolean autoClose)
+            throws IOException {
+        return makeSocketSafe(delegate.createSocket(s, host, port, autoClose), host);
+    }
+
+    @Override
+    public Socket createSocket(String host, int port) throws IOException {
+        return makeSocketSafe(delegate.createSocket(host, port), host);
+    }
+
+    @Override
+    public Socket createSocket(String host, int port, InetAddress localHost, int localPort)
+            throws IOException {
+        return makeSocketSafe(delegate.createSocket(host, port, localHost, localPort), host);
+    }
+
+    @Override
+    public Socket createSocket(InetAddress host, int port) throws IOException {
+        return makeSocketSafe(delegate.createSocket(host, port), host.getHostName());
+    }
+
+    @Override
+    public Socket createSocket(InetAddress address, int port, InetAddress localAddress,
+                               int localPort) throws IOException {
+        return makeSocketSafe(delegate.createSocket(address, port, localAddress, localPort),
+                address.getHostName());
+    }
+
+    private class CustomSSLSocket extends DelegateSSLSocket {
+
+        final boolean compatible;
+
+        @RequiresApi(api = Build.VERSION_CODES.Q)
```

Comment:
Is there anything in this constructor that requires Android Q (29)? If I'm reading this right, this app has a minSdkVersion of P (28); will this class still function as expected on devices running Android P? Or can either this `@RequiresApi` be dropped down to P or the minSdkVersion for the app updated to Q?

Reply:
Sorry. It's a temporary file. The file should not be included in this updates. WIll fix it.

Reply:
Thanks for the update; I don't see `CustomSocketFactory` in the pull request anymore, so marking this as resolved.